include/ht_scopedthread.h

```diff
@@ -14,22 +14,24 @@
 
 #pragma once
 
-#include <ht_platform.h>
-#include <thread>
+#include <ht_platform.h> //HT_API
+#include <ht_noncopy.h> //INonCopy
+#include <thread> //std::thread
 
-namespace Hatchit {
-   
-    namespace Core {
-        
-        class HT_API ScopedThread
+namespace Hatchit
+{   
+    namespace Core
+    {   
+        /**
+        \class ScopedThread
+        \ingroup HatchitCore
+        \brief A safety wrapper for std::threads to safely join threads
+        **/
```

Comment:
Please just move this doxygen block up under the copyright notice since that's the way it is in the other files